controllers/controllers_suite_test.go

```diff
@@ -142,6 +143,7 @@ var _ = BeforeSuite(func() {
 
 	Ω(infrav1.AddToScheme(scheme.Scheme)).Should(Succeed())
 	Ω(clusterv1.AddToScheme(scheme.Scheme)).Should(Succeed())
+	Ω(fakes.AddToScheme(scheme.Scheme)).Should(Succeed())
```

Comment:
So, I still don't understand the need for a `fakes` schema. I know you put a lot of work into this, but I think this was a bit of a miscommunication on my part. I think that just importing a machineset or equivalent would be completely sufficient here.

Reply:
for example, we use machineset, but it does not status.ready field. And we could not the logic when ready is false/true case.